src/uinput.rs

```diff
@@ -188,3 +232,93 @@ impl VirtualDevice {
         self.write_raw(&[syn])
     }
 }
+
+/// This struct is returned from the [VirtualDevice::enumerate_dev_nodes] function and will yield
+/// the syspaths corresponding to the virtual device. These are of the form `/dev/input123`.
+#[cfg(not(feature = "tokio"))]
+pub struct DevNodes {
+    dir: std::fs::ReadDir,
+}
+
+#[cfg(not(feature = "tokio"))]
+impl Iterator for DevNodes {
+    type Item = io::Result<PathBuf>;
+
+    fn next(&mut self) -> Option<Self::Item> {
```

Comment:
This looks like it could be condensed a bit with some fp techniques. Can you take another pass at it?